plan/physical_plan_builder.go

```diff
@@ -83,15 +83,17 @@ func (p *baseLogicalPlan) convert2NewPhysicalPlan(prop *requiredProp) (t task, e
 	}
 	// Now we only consider rootTask.
 	if len(p.basePlan.children) == 0 {
+		if !prop.isEmpty() {
```

Comment:
Add comments here. Here puzzled me why we can't put the non-empty `prop` to `genPhysPlansByReqProp`.
And we can judge `prop.isEmpty()` first.